source/extensions/filters/http/common/compressor/compressor.cc

```diff
@@ -506,9 +521,12 @@ bool CompressorFilterConfig::DirectionConfig::isMinimumContentLength(
     }
     return is_minimum_content_length;
   }
-
-  return StringUtil::caseFindToken(headers.getTransferEncodingValue(), ",",
-                                   Http::Headers::get().TransferEncodingValues.Chunked);
+  if (Runtime::runtimeFeatureEnabled(
+          "envoy.reloadable_features.enable_compression_without_content_length_header")) {
+    // returning true to account for HTTP/2 where content-length is optional
```

Comment:
and HTTP/1.1 chunked bodies, no?

Reply:
was this one overlooked?

Reply:
could you please elaborate on the last comment?. w/ this new diff we would compress even if there is no "transfer-encoding=chunked" header (if runtime is set to true) (aka there is no point to do separate check for chunked. because we would have logic like:
```
if true || chunked {
  return true;
}
```
this check would always be true
. if runtime set to false we would fall back to previous behavior  aka "return (chunked in headers?)"

Reply:
Yes, true.  Sorry, I meant to make the point that the comment implied that the return only applies to HTTP/2 but it applies to HTTP/1.1 without content length too.  How about something like

Return true to ignore the minimum length configuration if no content length is specified in the headers?